packages/bot-web-ui/src/components/dashboard/bot-builder/bot-builder.tsx

```diff
@@ -44,7 +44,12 @@ const BotBuilder = ({
     //removed used effect here because dark mode is an observable and the component will rerender
     setColors(is_dark_mode_on);
     previewRecentStrategy(selected_strategy_id);
-    if (active_tab === 1) loadFileFromRecent();
+
+    React.useEffect(() => {
+        if (active_tab === 1) {
+            loadFileFromRecent();
+        }
+    }, [is_dark_mode_on]);
```

Comment:
is dark mode on is an observable i think just keep dependency empty or add active_tab

Reply:
we need to run only when is_dark_mode_on change right? keeping it empty will run only once and keeping it outside will make it run always, which was causing this issue

Reply:
i think a reaction would be better right in load modal store... ? because keeping useeffect will cause 2 rerenders and performance issue  ?

Reply:
there is no 2 renders, re-render will happen only when updating state inside useEffect, so this function will be called only once even if we have inside or outside the useEffect

Reply:
<img width="1680" alt="image" src="https://user-images.githubusercontent.com/97010868/230329779-1eea7a34-195c-40ce-b8ee-2ae947d8bb48.png">
<img width="1680" alt="image" src="https://user-images.githubusercontent.com/97010868/230329837-c776bd2f-7e47-4901-a60f-97048509457b.png">

Reply:
i was talking about this render 
i think reaction is a good option here

Reply:
I did check there is no re-render, as no states are been changed